src/license_guc.h

```diff
@@ -13,61 +13,22 @@
 #include <export.h>
 #include <guc.h>
 
-/*
- * The license for using only Apache features is 'ApacheOnly'
- *
- * For documentation on other license types,
- * and information on the license format,
- * see tsl/src/license.c
- * (NB: This requires accepting LICENSE-TIMESCALE)
- */
-typedef enum LicenseType
-{
-	LICENSE_TYPE_APACHE_ONLY = 'A',
-	LICENSE_TYPE_COMMUNITY = 'C',
-	LICENSE_TYPE_ENTERPRISE = 'E',
-} LicenseType;
-
-#define TS_APACHE_ONLY_LICENSE "ApacheOnly"
-#define TS_COMMUNITY_LICENSE "CommunityLicense"
+#define TS_LICENSE_APACHE "apache"
+#define TS_LICENSE_TSL "tsl"
 
 /*
  * If compiled with APACHE_ONLY, default to using only Apache code.
  */
 #ifdef APACHE_ONLY
-#define TS_DEFAULT_LICENSE TS_APACHE_ONLY_LICENSE
+#define TS_LICENSE_DEFAULT TS_LICENSE_APACHE
 #else
-#define TS_DEFAULT_LICENSE TS_COMMUNITY_LICENSE
+#define TS_LICENSE_DEFAULT TS_LICENSE_TSL
 #endif
 
-#define TS_LICENSE_TYPE(license) license[0]
-
-#define TS_LICENSE_TYPE_IS_VALID(license)                                                          \
-	(TS_LICENSE_TYPE(license) == LICENSE_TYPE_APACHE_ONLY ||                                       \
-	 TS_LICENSE_TYPE(license) == LICENSE_TYPE_COMMUNITY ||                                         \
-	 TS_LICENSE_TYPE(license) == LICENSE_TYPE_ENTERPRISE)
-
-#define TS_LICENSE_IS_APACHE_ONLY(license) (TS_LICENSE_TYPE(license) == LICENSE_TYPE_APACHE_ONLY)
-
-#define TS_CURRENT_LICENSE_TYPE() TS_LICENSE_TYPE(ts_guc_license_key)
-
-#define TS_CURRENT_LICENSE_IS_APACHE_ONLY() TS_LICENSE_IS_APACHE_ONLY(ts_guc_license_key)
-/*
- * guc updating happens in two parts:
- *   1. The new guc value is validated, and any fallible code is run, but no
- *      externally-visible changes are performed
- *   2. The guc is set, and the externally-visible changes are performed
- *
- * This means that `ts_license_update_check` should not actually change anything
- * except for its `extra` parameter. (We cheat a little since in that we might
- * load a dynamic library during the `ts_license_update_check`, but we don't
- * consider that change to be visible.)
- */
-
-/* Each of these functions takes a LicenseUpdateExtra for their extra param */
-extern bool ts_license_update_check(char **newval, void **extra, GucSource source);
-extern void ts_license_on_assign(const char *newval, void *extra);
+extern bool ts_license_guc_check_hook(char **newval, void **extra, GucSource source);
+extern void ts_license_guc_assign_hook(const char *newval, void *extra);
 
-extern void TSDLLEXPORT ts_license_enable_module_loading(void);
+extern TSDLLEXPORT void ts_license_enable_module_loading(void);
+extern TSDLLEXPORT bool ts_license_is_apache(void);
```

Comment:
Are they called from TSL module?

Reply:
Only this one ts_license_enable_module_loading(). Removed export from the apache check function